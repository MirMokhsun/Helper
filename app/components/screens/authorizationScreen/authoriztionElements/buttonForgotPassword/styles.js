import { StyleSheet } from 'react-native';

export const styles = StyleSheet.create({
    wrapper: {
        marginBottom: 30,
        width: 290,
    },
    text: {
        fontSize: 14,
        color: "#1459D2",
        textAlign: 'right',
        fontFamily: 'Roboto-Light',
    },
});