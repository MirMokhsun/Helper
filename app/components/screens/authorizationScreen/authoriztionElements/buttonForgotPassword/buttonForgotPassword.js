import React from 'react';
import { View, TouchableOpacity, Text } from 'react-native';
import { withNavigation } from 'react-navigation';
import { styles } from './styles';

export class ButtonForgotPassword extends React.PureComponent {
    constructor(props) {
        super(props);
    }

    routeAuthorization = () => {
        const { navigation } = this.props;
        navigation.navigate('ForgotPasswordScreen')
    }

    render() {
        const { text } = this.props;
        return (
            <TouchableOpacity onPress={this.routeAuthorization} testID="ForgotPasswordAuthorizationID" accessibilityLabel="ForgotPasswordAuthorizationID">
                <View style={styles.wrapper}>
                    <Text style={styles.text}>{ text }</Text>
                </View>
            </ TouchableOpacity>
        )
    }
}

export default withNavigation(ButtonForgotPassword);