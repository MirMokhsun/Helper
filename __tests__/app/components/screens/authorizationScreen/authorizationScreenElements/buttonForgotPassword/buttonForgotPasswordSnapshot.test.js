import 'react-native';
import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';
import { ButtonForgotPassword } from '../../../../../../../app/components/screens/authorizationScreen/authoriztionElements/buttonForgotPassword/buttonForgotPassword';

it('create snapshot ButtonForgotPassword', () => {
    const renderer = new ShallowRenderer()
    let toolbarSnapshot = renderer.render(<ButtonForgotPassword text={"Forgot password?"} />);
    expect(toolbarSnapshot).toMatchSnapshot();
});