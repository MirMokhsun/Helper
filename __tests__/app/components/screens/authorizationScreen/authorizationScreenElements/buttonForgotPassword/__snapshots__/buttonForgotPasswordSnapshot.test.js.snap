// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`create snapshot ButtonForgotPassword 1`] = `
<TouchableOpacity
  accessibilityLabel="ForgotPasswordAuthorizationID"
  activeOpacity={0.2}
  onPress={[Function]}
  testID="ForgotPasswordAuthorizationID"
>
  <View
    style={
      Object {
        "marginBottom": 30,
        "width": 290,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#1459D2",
          "fontFamily": "Roboto-Light",
          "fontSize": 14,
          "textAlign": "right",
        }
      }
    >
      Forgot password?
    </Text>
  </View>
</TouchableOpacity>
`;
